packages/react/src/components/Table/TableHead/columnWidthUtilityFunctions.js

```diff
@@ -182,18 +190,31 @@ export const addMissingColumnWidths = ({ ordering, columns, currentColumnWidths
  */
 export const calculateWidthOnShow = (currentColumnWidths, ordering, colToShowIDs, columns) => {
   const visibleColumns = getVisibleColumns(currentColumnWidths, ordering, colToShowIDs);
-  const newColumnsToShow = colToShowIDs.reduce((accumulator, colToShowId) => {
+  const newWidthColumns = [];
+
+  const newColumnsToShow = colToShowIDs.map((colToShowId) => {
+    const existingWidth = getExistingColumnWidth(currentColumnWidths, columns, colToShowId);
     const widthOfColumnToShow =
-      getExistingColumnWidth(currentColumnWidths, columns, colToShowId) ||
-      getAverageVisibleColumnWidth(visibleColumns);
-    return [...accumulator, { id: colToShowId, width: Math.round(widthOfColumnToShow) }];
-  }, []);
+      existingWidth || getAverageVisibleColumnWidth(visibleColumns) || MIN_COLUMN_WIDTH;
+    const newColumnToShow = { id: colToShowId, width: Math.round(widthOfColumnToShow) };
+    if (!existingWidth) {
+      newWidthColumns.push(newColumnToShow);
+    }
+    return newColumnToShow;
+  });
+
   const totalWidthNeeded = newColumnsToShow.reduce((acc, col) => acc + col.width, 0);
   const shrinkableColumns = [...newColumnsToShow, ...visibleColumns].filter(
     (col) => col.width > MIN_COLUMN_WIDTH
   );
 
-  const adjustedCols = shrinkColumns(shrinkableColumns, totalWidthNeeded);
+  const adjustedCols = [
+    // There are some scenarios where the new columns don't have an existing width
+    // and in that case they are adjusted to get a min with assigned.
```

Comment:
Typo?
```suggestion
    // and in that case they are adjusted to get a min width assigned.
```

Reply:
Yes, thanks. I've made that mistake a few times.